awx/main/models/unified_jobs.py

```diff
@@ -1512,3 +1512,37 @@ def log_lifecycle(self, state, blocked_by=None):
         else:
             msg = f"{self._meta.model_name}-{self.id} {state.replace('_', ' ')}"
         logger_job_lifecycle.debug(msg, extra=extra)
+
+    @property
+    def launched_by(self):
+        ancestor_job = self.ancestor_job
+        attr = {
+            "manual" : "created_by",
+            "relaunch" : "created_by",
+            "scheduled" : "schedule" , 
+            "workflow" : "workflow", 
+            "webhook" : "job_template" , 
+            "sync" : "project" ,
+            "scm" : "inventory",
+            }
+
+        obj = getattr(ancestor_job, attr.get(ancestor_job.launch_type, ''), None)
+        if obj is not None:
+            return {
+                'id': obj.id,
+                'name': getattr(obj, 'name', None) or obj.username,
+                'type': obj.__class__.__name__,
```

Comment:
I believe that there is some standardized way we expose the type of objects in the API.  Let me see if I can't find it...

Reply:
Agreed, this is what we should use ☝️

Reply:
Updated the PR with ```get_type_for_model```